myapp/app/controllers/application_controller.rb

```diff
@@ -1,5 +1,6 @@
 class ApplicationController < ActionController::Base
   before_action :set_locale
+  before_action :redirect_to_maintenance
```

Comment:
By exempting specific actions from the before_action using the following approach, you can avoid checking the current path.

```suggestion
  before_action :redirect_to_maintenance, except: [:maintenance]
```